dev/fattest.simplicity/src/componenttest/topology/utils/MvnUtils.java

```diff
@@ -979,12 +981,64 @@ public static int runCmd(String[] cmd, File workingDirectory, File outputFile) t
         pb.redirectErrorStream(true);
 
         Log.info(c, "runCmd", "Running command " + Arrays.asList(cmd));
-        Process p = pb.start();
-        int exitCode = p.waitFor();
-        return exitCode;
+        
+        int hardTimeout;
+        
+        try {
+            hardTimeout = Integer.parseInt(System.getProperty("fat.timeout"));
+        }catch (NumberFormatException e) {
+            hardTimeout = 10800000;  // 10800000ms = 3hr
+        }
+        
+        long softTimeout = -1;
 
+        // We need to ensure that the hard timeout is large enough to avoid future issues
+        if (hardTimeout >= 30000) {
+            softTimeout = hardTimeout - 15000; // Soft timeout is 15 seconds less than hard timeout
+        }
+
+        Process p = pb.start();
+        int returnCode = 1;
+        boolean returnStatus;
+        if (softTimeout > -1) {
+            returnStatus = p.waitFor(softTimeout, TimeUnit.MILLISECONDS);  // Requires Java 8+
+            if (returnStatus == false) {
+                // Parse through the MVN logs for potential networking issues
+                if (outputFile.exists() && outputFile.canRead()) {
+                    try (Scanner s = new Scanner(outputFile)) {
+                        // Get the last two lines from the MVN log
+                        String lastLine = "";
+                        String secondLastLine = ""; 
+                        while (s.hasNextLine()) {
+                            secondLastLine = lastLine;
+                            lastLine = s.nextLine().toLowerCase();
+                        }
+                        // Check if the last or second line has the text "downloading" or "downloaded"
+                        // Throw custom timeout error rather then the one provided by the JUnitTask
+                        if ((lastLine.contains("downloaded") || lastLine.contains("downloading")) || (secondLastLine.contains("downloaded") || secondLastLine.contains("downloading"))) {       
+                            String timeoutMsg = "Timeout occurred. FAT timeout set to: " + hardTimeout + 
+                                                ". It appears there were some issues gathering dependencies. This may be due to network issues such as slow download speeds.";
+                            Log.info(c, "runCmd", timeoutMsg);  // Log the timeout message into messages.log or the default log 
+                            throw new AssertionFailedError(timeoutMsg);
+                        }
+                    } catch (FileNotFoundException FileError) {
+                        // Do nothing as we can't look at the MVN log. This leads to hard timeout handled by JUnitTask.
+                    }
+                }
+                // If the timeout is not for a reason we checked above then return to normal behavior and let it timeout through the Junit Task
+                returnCode = p.waitFor();
```

Comment:
I think we should remove this line and just let the test blow up with the soft timeout regardless of what keywords were found in the last few lines of the output

Reply:
Done. However this will still be called if the soft timeout can not be used.